eloquentjs/4_4_deep_comparison.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Performs deep comparison of two types and determines equality
+ * @param {*} itemA 
+ * @param {*} itemB 
+ * @returns 
+ */
+function deepEqual(itemA, itemB) {
+    if(itemA === null && itemB === null) {
+        return true;
+    }
+    if (typeof itemA === typeof itemB) {
```

Comment:
It's best to try to make your conditions as un-nested as possible. if typeof a !== typeof b you can return false from the whole function which lets you unnest the inner contents